rfcs/rfc-involuntary-dissolutions.md

```diff
@@ -0,0 +1,210 @@
+
+### Businesses
+
+A new **no_dissolution** field will be added to the **businesses** table.
+
+This field is intended to exclude a business from being selected as a candidate for the involuntary dissolution process.
+
+![Businesses Table](rfc-involuntary-dissolutions/business_table.png)
+
+## Involuntary Dissolutions Job
+
+Assumption:
+- Runs every day
+- Number of dissolutions that can be initiated per day will be retrieved from **configuration** table
+
+### 1. Initiating dissolution process for new businesses where AR has not been filed for 2 yrs and 2 months
+Use **businesses.last_ar_date** and other relevant business rules to determine candidate businesses that can enter dissolution process.
+
+1. Retrieve first **NUM_DISSOLUTIONS_ALLOWED** number of businesses to initiate the involuntary dissolution process for
+2. Create a new entry in **batches** table which will be used to uniquely identify the set of businesses that will be dissolved potentially if involuntary dissolution criteria is still met after specific duration of time has passed
+3. Create **batch_processing** entries to represent each business being dissolved and associate it to the batch created in the previous step.
+4. List of businesses that have dissolution process started are saved somewhere as csv.
+5. List of businesses that have dissolution process started are sent via email to a Staff accessible email address.
+
+### 2. Updating batch processing data for previously created in-progress batches
+1. Retrieve furnishing data for businesses that are currently in dissolution process.
+2. Update **batch_processing** entry data(**status**, **step**) as appropriate depending on time that has passed since dissolution process started, whether the appropriate furnishings have been sent and any other relevant conditions that need to be considered.
+
+### 3. Processing actual dissolution of businesses
+Assumption that following criteria has been met:
+* business is associated with one of the active dissolution batches.
+* business has received all furnishing messages
+* business still hasn’t fired the minimum ARs req’d to avoid dissolution process
+
+Retrieve all businesses tied to in process batches that can be dissolved and do the following for each business
+1. Dissolve business via an involuntary dissolution filing
+2. Update status for corresponding **batch_processing** entry.
+3. If all entries for corresponding batch have been processed, mark the batch as completed.  Note: furnishings job will take care of any final notifications that need to happen.
+
+Additional notes:
+* In the case where it is required to halt all new eligible businesses from entering the involuntary dissolution process, the **NUM_DISSOLUTIONS_ALLOWED** configuration will need to be set to zero.  Businesses that were already in the process of being dissolved before **NUM_DISSOLUTIONS_ALLOWED** was set to zero would continue the dissolution process.
```

Comment:
@thorwolpert you had mentioned that we need a way to halt dissolution runs.  Does this functionality cover what you were envisioning?

Reply:
Made updates to involuntary dissolution job section to clarify things.